### PR TITLE
[WIP][DO NOT REVIEW] Remove columns from DB for THREESCALE-2635

### PR DESCRIPTION
**Pull request by Martouta**
_Barely started_ :smile:

### CODE REVIEW

File: db/migrate/20190528133015_remove_service_support_emails.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class RemoveServiceSupportEmails < ActiveRecord::Migration
+  def change
+    safety_assured { remove_column :services, :tech_support_email, :string }
```
Comment (Martouta) [inline db/migrate/20190528133015_remove_service_support_emails.rb]: `ALTER TABLE `services` DROP `tech_support_email``

File: db/migrate/20190528133015_remove_service_support_emails.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    safety_assured { remove_column :services, :admin_support_email, :string }
```
Comment (Martouta) [inline db/migrate/20190528133015_remove_service_support_emails.rb]:
```sql
ALTER TABLE `services` DROP `admin_support_email`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class RemoveEndUserPlans < ActiveRecord::Migration
+  def change
+    safety_assured { remove_column :settings, :end_users_switch }
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
ALTER TABLE `settings` DROP `end_users_switch`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    safety_assured { remove_column :settings, :end_user_plans_ui_visible }
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
 ALTER TABLE `settings` DROP `end_user_plans_ui_visible`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+    safety_assured { remove_column :cinstances, :end_user_required }
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
ALTER TABLE `cinstances` DROP `end_user_required`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    safety_assured { remove_column :plans, :end_user_required }
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
ALTER TABLE `plans` DROP `end_user_required`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+    safety_assured { remove_column :services, :end_user_registration_required }
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
ALTER TABLE `services` DROP `end_user_registration_required`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    safety_assured { remove_column :services, :default_end_user_plan_id }
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
ALTER TABLE `services` DROP `default_end_user_plan_id`
```

File: db/migrate/20200211154433_remove_end_user_plans.rb [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+    drop_table :end_user_plans
```
Comment (Martouta) [inline db/migrate/20200211154433_remove_end_user_plans.rb]:
```sql
DROP TABLE `end_user_plans`
```

File: db/migrate/20200302170923_remove_accounts_deleted_at.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+class RemoveAccountsDeletedAt < ActiveRecord::Migration[5.0]
+  def change
+    return unless ActiveRecord::Base.connection.column_exists?(:accounts, :deleted_at)
+    remove_index :accounts, %i[state deleted_at]
```
Comment (Martouta) [inline db/migrate/20200302170923_remove_accounts_deleted_at.rb]:
```sql
DROP INDEX `index_accounts_on_state_and_deleted_at` ON `accounts`
```

File: db/migrate/20200302170923_remove_accounts_deleted_at.rb [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+    remove_index :accounts, %i[self_domain deleted_at]
```
Comment (Martouta) [inline db/migrate/20200302170923_remove_accounts_deleted_at.rb]:
```sql
DROP INDEX `index_accounts_on_self_domain_and_deleted_at` ON `accounts`
```

File: db/migrate/20200302170923_remove_accounts_deleted_at.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+    remove_index :accounts, %i[domain deleted_at]
```
Comment (Martouta) [inline db/migrate/20200302170923_remove_accounts_deleted_at.rb]:
```sql
DROP INDEX `index_accounts_on_domain_and_deleted_at` ON `accounts`
```

File: db/migrate/20200302170923_remove_accounts_deleted_at.rb [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+    safety_assured { remove_column :accounts, :deleted_at }
```
Comment (Martouta) [inline db/migrate/20200302170923_remove_accounts_deleted_at.rb]:
```sql
ALTER TABLE `accounts` DROP `deleted_at`
```

File: db/migrate/20190913002814_remove_service_preffix_from_accounts.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+class RemoveServicePreffixFromAccounts < ActiveRecord::Migration
+  def change
+    safety_assured { remove_column :accounts, :service_preffix, :string }
```
Comment (Martouta) [inline db/migrate/20190913002814_remove_service_preffix_from_accounts.rb]:
```sql
ALTER TABLE `accounts` DROP `service_preffix`
```

**Comment by Martouta** [comment]
**WIP - RUNNING IN PREVIEW**

05:02 deploy:migrate
      [deploy:migrate] Run `rake db:migrate`
      01 ruby2.4 /usr/local/bin/bundle exec rake2.4 db:migrate
      01 DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from requir…
      01 RAILS_ENV=preview environment is not defined in config/webpacker.yml, falling back to production environment
      01 == 20200326154943 RemoveServiceSupportEmails: migrating =======================
      01 -- remove_column(:services, :tech_support_email, :string)
      01    -> 4.8649s
      01 -- remove_column(:services, :admin_support_email, :string)
      01    -> 4.7415s
      01 == 20200326154943 RemoveServiceSupportEmails: migrated (9.6068s) ==============
      01
      01 == 20200326160315 RemoveAccountsDeletedAt: migrating ==========================
      01 -- column_exists?(:accounts, :deleted_at)
      01    -> 0.0037s
      01 -- index_exists?(:accounts, [:state, :deleted_at])
      01    -> 0.0010s
      01 -- index_exists?(:accounts, [:self_domain, :deleted_at])
      01    -> 0.0009s
      01 -- index_exists?(:accounts, [:domain, :deleted_at])
      01    -> 0.0009s
      01 -- remove_column(:accounts, :deleted_at)
      01    -> 554.7010s
      01 == 20200326160315 RemoveAccountsDeletedAt: migrated (554.7102s) ===============
      01
      01
      01 == 20200326164004 RemoveServicePreffixFromAccounts: migrating =================
      01 -- remove_column(:accounts, :service_preffix, :string)

**Comment by hallelujah** [comment]
How long does that take in total? So we can estimate the downtime?

**Comment by Martouta** [comment]
@hallelujah it is still running 😄 when it is done, I will update it here. have a bit of patience 😝

**Comment by Martouta** [comment]
Closing for now, to don't overcomplicate things, but I will reopen soon (hopefully today)

**Pull request closed without merging**